### ISSUE TITLE
Our issues with Reviewable

### ISSUE THREAD

**Issue by joelmccracken** [Reviewable/Reviewable#112]
Hi,

We just tried using Reviewable for a while, and quickly decided it isn't for us. A few notes:

1. We found the review UI confusing. Not sure what exactly to say about it, except that it seemed very unintuitive, and e.g. the skumorphic paper tear was distracting.

2. The add-your-button-to-our-pr-descriptions feature clobbered some of our pr descriptions as they were being edited. It happened to two of us in the span of a few hours. 

3. We *really* want to be able to review commit by commit, as our team practices good commit discipline. 

Our misc wish list of features we really want out of such a service:
https://gist.github.com/carols10cents/abc5628132a55a647214

**Comment by pkaminski** [comment]
Thanks for the feeback Joel, it's very valuable -- I'm sorry Reviewable didn't work out for your team even though it satisfies all the requirements on your wishlist.

1. Perceptions of the UI vary widely and I have yet to figure out what affects this.  Besides the torn paper, are there other specific elements you can point to?  Do you think an interactive onboarding walkthrough might've helped?

2. That's terrible, my apologies!  I'll adjust the code to minimize the window of opportunity for racing edits, but AFAICT there's no way to make the edit atomic (confirming with GitHub support).  :disappointed: 

3. This is possible: take a look at #77.  The TL;DR is that you can either add a "review each commit" label to the PR, or request that I flip a repo-wide flag for you.  (The latter mechanism is still experimental, hence why there's no self-serve UI for it yet, but Servo has been using it successfully for over a month.)

The two extra comments on your wishlist are interesting -- they're things I'm looking into adding to Reviewable in the long term, but they're a bit more challenging than the more common review functionality.

**Comment by carols10cents** [comment]
> it satisfies all the requirements on your wishlist.

Oh! I must be missing the next/prev commit buttons? I tried enabling the "review by commit" functionality as described in #77, and I expected that to change the interface to show me the first commit in the PR with a button to go to the next commit in the PR, but it still looks like the navigation is by file? That was confusing to me.

In general, the UI is just pretty cluttered; I was overwhelmed with options and things to do at the top and had to scroll down quite a ways to get to what I really wanted, which was to see the code I'm reviewing.

**Comment by carols10cents** [comment]
Also, could reviewable add a comment to a PR instead of editing the description? That would get rid of the edit conflict issues...

**Comment by joelmccracken** [comment]
Re: 2, is there a way to disable adding those buttons to the PR? Or, add it as a comment to the PR, instead?

**Comment by joelmccracken** [comment]
ugh @carols10cents totally beat me to it.

**Comment by pkaminski** [comment]
@carols10cents, there's no explicit next/prev commit buttons (yet), but you can get the same effect with one click:  in the file matrix header, drag from base-r1 to get diffs for the first commit, r1-r2 for diffs for the second commit, etc.  (As long as you have "review each commit" label set on the PR, or asked me to turn on the equivalent repo flag on the backend.)  The per-commit-review workflow is still pretty new to Reviewable so I'm in the process of building out better explicit support for it as you can see in #77.

As for the Reviewable button in the PR, I think it's important to have a clear link to the review so that people can find it easily, otherwise some will keep leaving comments in GitHub which is [not ideal](https://github.com/Reviewable/Reviewable/wiki/FAQ#can-i-mix-using-github-and-reviewable-on-a-pr).  And for the sake of consistency, I think it's better to have the button always in the same place, rather than in a random comment that could be some way down the page.  I believe the edit race condition should be fine now -- I'd let things slip so that there was a window of many seconds (or dozens of seconds, for large reviews!) between the GET and the PATCH.  That window is now back in the milliseconds range so I'm hoping Reviewable won't clobber any more edits.  But if it does happen again, I'll certainly reconsider.

**Comment by joelmccracken** [comment]
I have more to add to this at some point, but I will say that we are all confused by the review matrix. A screencast/intro on how to use reviewable would be a great idea.

**Comment by pkaminski** [comment]
Understood, and thank you for the feedback -- it tallies pretty well with what I heard from others, too.  Production of a screencast is under way, but it's intended more for marketing (*why* you might want to use Reviewable) than teaching, though it will show some common flows.  I'm also making progress on an interactive onboarding / help system that will ease new users into the UI and workflow.  Finally, I'm strongly considering showing a simpler UI focused on the common use cases by default and revealing the full file matrix only on demand, for power users.

In other words, one way or another this will get solved, though it might take a week or two.  :smile:

**Comment by pkaminski** [comment]
I believe that the new release up now should address your concerns:

- The file matrix is hidden by default and a new UI guides you through the review workflow.  If you do open the file matrix, the legend is much easier to find.
- You can switch a review, or an entire repo, to review-each-commit mode.
- There's now explicit buttons for marking a commit as reviewed, and for advancing the diff to the next commit.
- The skeuomorphic torn paper is gone.
- PR descriptions shouldn't be getting clobbered any more.

Better onboarding and a screencast are still coming later.

**Comment by pkaminski** [comment]
I'll close this issue now, but if you're still interested in Reviewable and the changes above aren't sufficient, please reopen!
